services-js/registry-certs/client/birth/components/SupportingDocumentsInput.tsx

```diff
@@ -0,0 +1,276 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import {
+  CloseButton,
+  FOCUS_INDICATOR_COLOR,
+  VISUALLYHIDDEN,
+} from '@cityofboston/react-fleet';
+
+interface Props {
+  name: string;
+  title: string;
+  fileTypes?: string[] | string;
+  sizeLimit: FileSize;
+  handleChange(documents: File[]): void;
+}
+
+interface State {
+  selectedFiles: File[];
+  isFocused: boolean;
+}
+
+interface FileSize {
+  amount: number;
+  unit: 'B' | 'KB' | 'MB' | 'GB';
+}
+
+/**
+ * Component that allows a user to select one or more files from their computer.
+ *
+ * Selected files can be individually removed, and additional files can be
+ * added to the group of already-selected files. Duplicate files are identified
+ * by filename, and will be discarded.
+ *
+ * Defaults to accept any file type or size. To restrict the types of files to
+ * accept, assign a string or array of strings to “fileTypes” that contains the
+ * allowed MIME type(s).
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#Limiting_accepted_file_types
+ *
+ * To disallow large files, pass an object to “sizeLimit” to define the limit:
+ *
+ * { amount: 20, unit: 'MB' }
+ */
+export default class SupportingDocumentsInput extends React.Component<
+  Props,
+  State
+> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      selectedFiles: [],
+      isFocused: false,
+    };
+  }
+
+  private readonly inputRef = React.createRef<HTMLInputElement>();
+
+  // Check to see if a file does not exceed the file size limit, if defined.
+  private checkFileSize = (file: File): boolean | void => {
+    const sizeLimit = handleBytes.convert(this.props.sizeLimit);
+    const fileSize = handleBytes.format(file.size);
+
+    if (file.size > sizeLimit) {
+      // todo: consider handling via callback: https://github.com/CityOfBoston/digital/pull/41#discussion_r218545231
+
+      alert(
```

Comment:
TODO: don't use `alert`

I also wonder about returning `false` from a file-upload callback to tell the picker to throw away the selection.

Reply:
There are a couple of todos regarding this; it’s definitely something I’d like to come back to later.